File: core/src/main/java/com/tek/core/properties/TekCorsProperties.java
package com.tek.core.properties;

import java.util.stream.Stream;
import lombok.Data;
import org.springframework.web.bind.annotation.RequestMethod;

/**
 * Configuration properties for CORS Filter.
 * <p>
 * Fallback to default configuration if none provided.
 *
 * @author MarcoPagan
 */
@Data
public class TekCorsProperties {

  private String allowedOrigin = "http://localhost:4200";
  private Boolean allowedCredentials = true;
  private String[] allowedMethods =
      Stream.of(RequestMethod.values()).map(RequestMethod::name).toArray(String[]::new);
  private String[] allowedHeaders =
      Stream.of(RequestHeader.values()).map(RequestHeader::label).toArray(String[]::new);

  enum RequestHeader {

    X_REQUESTED_WITH("x-requested-with"),
    L_AUTHORIZATION("authorization"),
    CONTENT_TYPE("Content-Type"),
    U_AUTHORIZATION("Authorization"),
    CREDENTIAL("credential"),
    X_XSRF_TOKEN("X-XSRF-TOKEN");

    private final String label;

    RequestHeader(String label) {
      this.label = label;
    }

    public String label() {
      return this.label;
    }
  }
}

File: security/src/main/kotlin/com/tek/security/configuration/TekCorsProperties.kt
package com.tek.security.configuration

import org.springframework.boot.autoconfigure.security.SecurityProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

/**
 * Configuration lookup for [TekCorsFilter].
 *
 * Default configuration is provided if none is found (development purpose only)
 *
 */
@Configuration
@ConfigurationProperties(prefix = "security.oauth2.cors")
@EnableConfigurationProperties(SecurityProperties::class)
class TekCorsProperties {

    var allowedOrigin: String = "http://localhost:4200"

    var allowedCredentials: String = "true"

    var allowedMethods: Array<String> = arrayOf("GET", "POST", "PATCH", "PUT", "DELETE", "OPTIONS")

    var allowedHeaders: Array<String> =
        arrayOf("x-requested-with", "authorization", "Content-Type", "Authorization", "credential", "X-XSRF-TOKEN")
}